### PR TITLE
refactor(net): feature flag non tcp APIs

### PR DESCRIPTION
**Pull request by lucacasonato**

### CODE REVIEW

File: extensions/net/io.rs [hunk 1]
```diff
@@ -219,11 +209,17 @@ async fn op_shutdown(
     .resource_table
     .get_any(rid)
     .ok_or_else(bad_resource_id)?;
-  if let Some(s) = resource.downcast_rc::<TcpStreamResource>() {
+  #[cfg(all(unix, feature = "full"))]
+  if let Some(s) = resource.downcast_rc::<UnixStreamResource>() {
     s.shutdown().await?;
```
Comment (piscisaureus) [inline extensions/net/io.rs:214]:
```suggestion
    return s.shutdown().await;
```

File: extensions/net/io.rs [hunk 2]
```diff
@@ -219,11 +209,17 @@ async fn op_shutdown(
...
-  } else if let Some(s) = resource.downcast_rc::<TlsStreamResource>() {
+    return Ok(());
+  }
+  #[cfg(feature = "full")]
+  if let Some(s) = resource.downcast_rc::<TlsStreamResource>() {
     s.shutdown().await?;
```
Comment (piscisaureus) [inline extensions/net/io.rs:219]:
```suggestion
    return s.shutdown().await;
```

**Comment by piscisaureus** [comment]
Have you considered using `cfg_if` instead of adding boatloads of `[cfg]` attributes? It is used something like this:


```rust
cfg_if::cfg_if! {
    if #[feature = "full"] {
      use foo:foo;
      use std::bar:xyz;
    }
}
```

**Comment by lucacasonato** [comment]
@piscisaureus Oh that looks a lot better indeed. I am discussing with Bartek how we could prevent this feature flag from being needed at all (at least for the time being). If the conclusion is that we need the flag anyway, I'll look into `cfg_if`.

**Comment by piscisaureus** [comment]
Might be easier to just feature gate the op registration logic.

**Comment by lucacasonato** [comment]
I am going to solve this differently.

**Pull request closed without merging**